### ISSUE TITLE
PanResponder crashing + not working after translate View

### ISSUE THREAD

**Issue by DanielSRS** [microsoft/react-native-macos#2214]
### Environment

```markdown
react-native -v: 0.75.3
npm ls react-native-macos: react-native-macos@0.75.7
node -v: v21.7.1
npm -v: 10.5.0
yarn --version: 3.6.4
xcodebuild -version: Xcode 15.4 Build version 15F31d
```

### Steps to reproduce the bug

1. Copy and run the PanResponder example from react native [docs](https://reactnative.dev/docs/0.74/panresponder)
2. Drag the blue square around
3. Release the mouse
4. Try drag the box again

### Expected Behavior

- I expected to drag the View around as many times I want and to any position
- Be able to set useNativeDrive to true
- Restart drag from the last position I released it

Example working on mobile: 

https://github.com/user-attachments/assets/21986dd0-a667-4e0e-b32f-0f4ef4a92e6b

### Actual Behavior

- I can drag the view around as long as I do not release the mouse
- If I release the mouse I cannot restart dragging the view again from its last position, but if I try to do it from its initial position it somewhat works
- When setting useNativeDrive to true, the app crashes with an uncaught exception

With useNativeDrive is false, it partially works, but with this dragging issue:

https://github.com/user-attachments/assets/c14a84e2-f1a1-4ba5-87b3-b12a896aa724

Error I got after setting useNativeDrive to true:
```
TypeError: config.onPanResponderMove is not a function. (In 'config.onPanResponderMove(event, gestureState)', 'config.onPanResponderMove' is an instance of AnimatedEvent)
```
![Screenshot 2024-10-07 at 02 49 34](https://github.com/user-attachments/assets/2e1552b1-ef14-4b36-9ae1-4cc777b88db2)

### Reproducible Demo

I took the code directly from React Native website [here](https://reactnative.dev/docs/panresponder), but since the code is really small, I will just past it here bellow

<details>
<summary>Example code</summary>

```Typescript
import React, {useRef} from 'react';
import {Animated, View, StyleSheet, PanResponder, Text} from 'react-native';

const App = () => {
  const pan = useRef(new Animated.ValueXY()).current;

  const panResponder = useRef(
    PanResponder.create({
      onMoveShouldSetPanResponder: () => true,
      onPanResponderMove: Animated.event([null, {dx: pan.x, dy: pan.y}], { useNativeDriver: true }),
      onPanResponderRelease: () => {
        pan.extractOffset();
      },
    }),
  ).current;

  return (
    <View style={styles.container}>
      <Text style={styles.titleText}>Drag this box!</Text>
      <Animated.View
        style={{
          transform: [{translateX: pan.x}, {translateY: pan.y}],
        }}
        {...panResponder.panHandlers}>
        <View style={styles.box} />
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  titleText: {
    fontSize: 14,
    lineHeight: 24,
    fontWeight: 'bold',
  },
  box: {
    height: 150,
    width: 150,
    backgroundColor: 'blue',
    borderRadius: 5,
  },
});

export default App;
```

</details>

### Additional context

- This happens in react-native-macos 0.73 too, except it doesn't crash when setting useNativeDrive to true;
- I didn't test with 0.74, but I will assume It behaves the same as 0.73 and 0.75
- I tested this with Blank new initialized projects

Initially I thought it was an issue with PanResponder alone, but searching a bit, it may be related with:
- This issue #2147
- And maybe this one #1099 that seems to have been closed by the bot

**Comment by Saadnajmi** [comment]
Well, this is indeed a bug I have not seen before. Thanks for the report! I'm not sure when we'll look at it, but I wanted you to know I've seen your issue

**Comment by DanielSRS** [comment]
Yeah, I can see why thats not a priority. Is not difficult work to around it, so it's not a big problem. Thanks for letting me know, I really appreciate it 😁
